Source/VCGeneration/SplitAndVerifyWorker.cs

```diff
@@ -160,6 +165,65 @@ private async Task ProcessResult(Split split, CancellationToken cancellationToke
       await HandleProverFailure(split, cancellationToken);
     }
 
+    private static bool IsProverFailed(ProverInterface.Outcome outcome)
+    {
+      switch (outcome)
+      {
+        case ProverInterface.Outcome.Valid:
+        case ProverInterface.Outcome.Invalid:
+        case ProverInterface.Outcome.Undetermined:
+          return false;
+        case ProverInterface.Outcome.OutOfMemory:
+        case ProverInterface.Outcome.TimeOut:
+        case ProverInterface.Outcome.OutOfResource:
+          return true;
+        default:
+          Contract.Assert(false);
+          throw new cce.UnreachableException();
+      }
+    }
+
+    private static Outcome MergeOutcomes(Outcome currentOutcome, ProverInterface.Outcome newOutcome)
+    {
+      switch (newOutcome)
+      {
+        case ProverInterface.Outcome.Valid:
+          return currentOutcome;
+        case ProverInterface.Outcome.Invalid:
+          return Outcome.Errors;
+        case ProverInterface.Outcome.OutOfMemory:
+          if (currentOutcome != Outcome.Errors && currentOutcome != Outcome.Inconclusive)
+          {
+            return Outcome.OutOfMemory;
+          }
+
+          return currentOutcome;
+        case ProverInterface.Outcome.TimeOut:
+          if (currentOutcome != Outcome.Errors && currentOutcome != Outcome.Inconclusive)
+          {
+            return Outcome.TimedOut;
+          }
+
+          return currentOutcome;
+        case ProverInterface.Outcome.OutOfResource:
+          if (currentOutcome != Outcome.Errors && currentOutcome != Outcome.Inconclusive)
+          {
+            return Outcome.OutOfResource;
+          }
+
+          return currentOutcome;
+        case ProverInterface.Outcome.Undetermined:
+          if (currentOutcome != Outcome.Errors)
+          {
+            return Outcome.Inconclusive;
+          }
+          return currentOutcome;
+        default:
+          Contract.Assert(false);
+          throw new cce.UnreachableException();
+      }
+    }
+
```

Comment:
Thanks for moving these to separate methods. I think I'll want to use them to resolve #494.